src/main/java/Duke.java

```diff
@@ -9,57 +11,101 @@ public static void main(String[] args) {
+                } else {
                     tasks[taskNum - 1].setDone(true);
                     System.out.println("Nice! I've marked this task as done: ");
                     tasks[taskNum - 1].printTask();
                 }
-            }
-            else if (str.equals("unmark")){
-                printTasks(tasks, numTasks);
+            } else if (str.equals("unmark")) {
+                printTasks();
                 System.out.println("Which task would you like to mark as incomplete?");
                 taskNum = sc.nextInt();
                 sc.nextLine();
-                if (taskNum > numTasks || taskNum <= 0){
+                if (taskNum > numTasks || taskNum <= 0) {
                     System.out.println("Uh oh! It seems like that task doesn't exist!");
-                }
-                else {
+                } else {
                     tasks[taskNum - 1].setDone(false);
                     System.out.println("Okie, I've marked this task as not done yet:");
                     tasks[taskNum - 1].printTask();
                 }
+            } else if (str.equals("add")) {
+                System.out.println("Okie, what type of task is this?");
+                System.out.println("1.Todo");
+                System.out.println("2.Deadline");
+                System.out.println("3.Event");
+                str = sc.nextLine();
+                switch (str) {
+                case "1":
+                    addTodo();
+                    break;
+                case "2":
+                    addDeadline();
+                    break;
+                case "3":
+                    addEvent();
+                    break;
+                default:
+                    System.out.println("Uh oh, please enter a valid number!");
+                }
+                System.out.println("______________________________________");
             }
-            else {
-                tasks[numTasks] = new Task(str);
-                numTasks++;
-                System.out.println("added " + str);
+            else{
+                System.out.println("Sorry, I don't recognise that command. Please try again!");
+                System.out.println("______________________________________");
             }
-            System.out.println("______________________________________");
             str = sc.nextLine();
         }
         System.out.println("Bye! Hope to see you again!");
     }
 
-    public static void printTasks(Task[] tasks, int numTasks){
+    public static void printTasks(){
         for (int i=0; i<numTasks; i++){
             System.out.print((i+1) + ". ");
             tasks[i].printTask();
         }
     }
+
+    public static void addTodo(){
+        Scanner sc = new Scanner(System.in);
+        System.out.println("Okie, what should I call the task?");
+        String str = sc.nextLine();
+        tasks[numTasks] = new Todo(str);
+        numTasks++;
+        System.out.println("Added!");
+    }
```

Comment:
All of your methods look great! Perhaps you could consider adding Javadoc comments to them?